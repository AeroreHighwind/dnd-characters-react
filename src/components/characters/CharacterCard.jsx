import React from 'react';
import CharacterPortrait from './CharacterPortrait';
import { CharacterType } from "../../core/types/CharacterCardTypes";

export default function CharacterCard({ character }) {
  return (
    <div className='flex flex-col justify-center overflow-hidden max-w-md'>
      <p className='text-lg font-bold overflow-hidden text-ellipsis whitespace-nowrap'>
        {character.name}
      </p>
      <div className='flex bg-gray-700 h-52 w-52 text-slate-950 justify-center items-center'>
        <img
          className='w-full h-full'
          src={character.imgUrl}
          alt="Character Portrait"
          onError={(e) => {
            // @ts-ignore
            e.target.src = "unknown.png";
          }}
        />
    </div>
      {/* <CharacterPortrait imgUrl={character.imgUrl} /> */}
      <div>
      <h1 className='font-sans font-semibold'>Lvl: {character.lvl} / <span className='text-lg'>{character.class}</span></h1>
    </div>
    </div>
  );
}

CharacterCard.propTypes =  {
  character: CharacterType.isRequired,
};;

