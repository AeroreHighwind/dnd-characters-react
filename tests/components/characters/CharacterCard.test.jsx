import { render } from "@testing-library/react";
import CharacterCard from "../../../src/components/characters/CharacterCard";
import React from "react";
import { mockedCharacter } from "../../__mocks__/character.mock";

describe("Testing character card component", () => {
  let testCharacter;
  let containerRef;

  beforeEach(()=>{
     testCharacter = mockedCharacter
    const { container } = render(<CharacterCard character={testCharacter} />);
    containerRef = container;
  })

  test("should match with snapshot", () => {
    expect(containerRef).toMatchSnapshot();
  });

  test('should show the caracter name in a p element', () => { 
    const p = containerRef.querySelector('p')
    expect(p.innerHTML).toContain(testCharacter.name)
   })
});