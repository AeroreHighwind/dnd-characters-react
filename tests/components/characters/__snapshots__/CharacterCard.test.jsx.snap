// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing character card component should match with snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-center overflow-hidden max-w-md"
  >
    <p
      class="text-lg font-bold overflow-hidden text-ellipsis whitespace-nowrap"
    >
      Test Character
    </p>
    <div
      class="flex bg-gray-700 h-52 w-52 text-slate-950 justify-center items-center"
    >
      <img
        alt="Character Portrait"
        class="w-full h-full"
        src="mockedurl.test.com"
      />
    </div>
    <div
      class="font-semibold"
    >
      <h1
        class="font-sans"
      >
        Lvl: 
        10
         / 
        <span
          class="text-lg"
        >
          Tester
        </span>
      </h1>
      <h2>
        Series: 
        unit testing
      </h2>
    </div>
  </div>
</div>
`;
